lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -5433,6 +5434,11 @@ void ByteCodeGenerator::EmitPropLoad(Js::RegSlot lhsLocation, Symbol *sym, Ident
             case Js::PropertyIds::undefined:
                 opcode = Js::OpCode::LdUndef;
                 break;
+            case Js::PropertyIds::__chakraLibrary:
+                if (CONFIG_FLAG(LdChakraLib)) {
+                    opcode = Js::OpCode::LdChakraLib;
```

Comment:
>opcode = Js::OpCode::LdChakraLib; [](start = 20, length = 33)

Maybe I didn't see the logic, but I think after the first time you run this bytecode, a flag could be set so that any future generated JIT does not need to run this instruction to load the lib. The helper should also ensure it is only loaded once.